mlir/lib/Target/LLVM/NVVM/Target.cpp

```diff
@@ -473,6 +473,18 @@ NVPTXSerializer::compileToBinary(const std::string &ptxCode) {
     }                                                                          \
   } while (false)
 
+#include <nvFatbin.h>
+
+#define RETURN_ON_NVFATBIN_ERROR(expr)                                         \
+  do {                                                                         \
+    auto result = (expr);                                                      \
+    if (result != nvFatbinResult::NVFATBIN_SUCCESS) {                          \
+      emitError(loc) << llvm::Twine(#expr).concat(" failed with error: ")      \
+                     << nvFatbinGetErrorString(result);                        \
+      return std::nullopt;                                                     \
+    }                                                                          \
+  } while (false)
```

Comment:
Macros are heavily discouraged in MLIR. Please try to find a better solution, for example by wrapping this into some lambda helper or something.

Reply:
There is `RETURN_ON_NVPTXCOMPILER_ERROR` just above this code...

Since `#expr` gives the user more information, is it acceptable to move this macro into the function?

Reply:
I didn't see that, my bad. I'll delegate this to @grypp as my comments were just side remarks.

Reply:
I hate macros, but for interfacing with this kind of library, that may still be the cleanest right now.